sw/device/silicon_creator/lib/ownership/isfb.c

```diff
@@ -0,0 +1,133 @@
+  kExpectedStrikeBitCount = 256,
+
+  // The expected size of the strike region in bytes and words.
+  kExpectedStrikeRegionBytesCount = kExpectedStrikeBitCount * sizeof(uint32_t),
+  kExpectedStrikeRegionWordsCount =
+      kExpectedStrikeRegionBytesCount / sizeof(uint32_t),
+};
+
+rom_error_t isfb_boot_request_process(const manifest_ext_isfb_t *ext,
+                                      const owner_config_t *owner_config,
+                                      uint32_t *checks_performed_count) {
+  *checks_performed_count = UINT32_MAX;
+  if (ext->header.name != kManifestExtIdIsfb ||
+      (hardened_bool_t)owner_config->isfb == kHardenedBoolFalse) {
+    return kErrorOwnershipISFBNotPresent;
+  }
+  const owner_isfb_config_t *isfb = owner_config->isfb;
+  if (ext->product_expr_count * 2 > isfb->product_words ||
+      ext->product_expr_count > kProductExprMaxCount) {
+    return kErrorOwnershipISFBProductExpCnt;
+  }
+
+  flash_ctrl_perms_t perm = {
+      .read = kMultiBitBool4True,
+      .write = kMultiBitBool4True,
+      .erase = kMultiBitBool4False,
+  };
+  flash_ctrl_cfg_t cfg = {
+      .scrambling = kMultiBitBool4False,
+      .ecc = kMultiBitBool4True,
+      .he = kMultiBitBool4False,
+  };
+
+  flash_ctrl_info_page_t isfb_info_page;
+  HARDENED_RETURN_IF_ERROR(flash_ctrl_info_type0_params_build(
+      isfb->bank, isfb->page, &isfb_info_page));
+  flash_ctrl_info_perms_set(&isfb_info_page, perm);
+  flash_ctrl_info_cfg_set(&isfb_info_page, cfg);
+
+  // There are in total 128 bits in the strike mask. Each bit corresponds to a
+  // flash word. Each flash word is 64 bits.
+  uint32_t data[kExpectedStrikeRegionWordsCount];
+  static_assert(sizeof(ext->strike_mask) * CHAR_BIT == kExpectedStrikeBitCount,
+                "Strike mask size mismatch");
+  static_assert(sizeof(data) == kExpectedStrikeRegionBytesCount,
+                "Data size mismatch");
+
+  HARDENED_RETURN_IF_ERROR(flash_ctrl_info_read(&isfb_info_page, /*offset=*/0,
+                                                ARRAYSIZE(data), data));
+
+  uint32_t strike_cnt_ok = 0;
+  uint32_t strike_cnt_bad = 0;
+  uint32_t *strike_word = (uint32_t *)data;
+  size_t i;
+  for (i = 0; i < ARRAYSIZE(data); ++i, ++strike_word) {
+    uint32_t strike_bit = ext->strike_mask[i >> 5] & (1 << (i & 31));
+    if (launder32(strike_bit) && *strike_word != UINT32_MAX) {
+      strike_cnt_bad++;
+    } else if (!launder32(strike_bit) || *strike_word == UINT32_MAX) {
+      strike_cnt_ok++;
+    } else {
+      // This condition is equivalent to a potential fault in either the strike
+      // mask or the ISFB info buffer.
+      strike_cnt_bad++;
+    }
+  }
+  // Check loop completion and count consistency.
+  HARDENED_CHECK_EQ(strike_cnt_ok + strike_cnt_bad + i,
+                    kExpectedStrikeBitCount * 2);
+
+  if (launder32(strike_cnt_bad) > 0) {
+    return kErrorOwnershipISFBStrikeMask;
+  }
+
+  static_assert(sizeof(data) == KProductExprMaxCountBytes,
```

Comment:
This doesn't seem like the correct comparisons. The data should be the size of the rollback space not the max product space (even though they happen to be the same right now)

Reply:
There is a separate check for the strikes region above. I will update this check again once we have the updated memory map you recommended in a separate comment.

Reply:
This check is no longer here as I updated the info flash read to load the entire page.